js/markDone.js

```diff
@@ -0,0 +1,30 @@
+import { EDIT, COMPLETE } from "/js/constants.js";
+import calculateDuration from "/js/utility/mDDuration.js";
+import createDurationElement from "/js/utility/mDDuElement.js";
+
+export default function markTodoAsCompleted(element, taskList, id) {
+  const index = taskList.findIndex((item) => item.id === id);
+  if (index === -1) {
+    return;
+  }
+  taskList[index].done = true;
+
+  const text = element.querySelector(".text").style;
+  text.textDecoration = "line-through";
+  text.color = "green";
+
+  element.querySelector(`[data-job="${COMPLETE}"]`).style.display = "none";
+  element.querySelector(`[data-job="${EDIT}"]`).style.display = "none";
+
+  const taskStartDay = element.querySelector(".date").innerText;
+  let taskStartDateString = taskStartDay.replace(/[^0-9]/g, "-");
+  taskStartDateString = taskStartDateString.slice(12);
+
+  const startDate = new Date(taskStartDateString).getTime();
```

Comment:
do not trust DOM content. consider passing the `taskStartDate` to the function as argument to avoid DOM access and data cleanup.

Reply:
I have passed the taskStartDate as an argument of calculateDuration function. and convert string to date format inside the calculateDuration function. please check

Reply:
what I meant is that the function `markTodoAsCompleted` should take the `taskStartDate` argument instead and avoid DOM access for that data.

anyway, don't you have access to the `startDate` from the `task` itself? e.g `task.startDate`